_clients/index.md

```diff
@@ -35,14 +35,19 @@ OpenSearch provides clients for the following programming languages and platform
 * **Rust**
   * [OpenSearch Rust client]({{site.url}}{{site.baseurl}}/clients/rust/)
 
-All clients are compatible with any version of OpenSearch.
+All clients are compatible with any version of OpenSearch. However, major versions of OpenSearch introduce breaking changes that require careful upgrades of the clients. For example, while clients 2.0.0 work against the latest OpenSearch 1.x, certain deprecated features removed in OpenSearch 2.0 have also been removed from the clients. For a client compatibility matrix, see the COMPATIBILITY.md file in the client's repository.
 {: .note}
 
+The OpenSearch Java high-level REST client will be deprecated starting with OpenSearch version 3.0.0 and will be removed in a future release. We recommend switching to the [Java client]({{site.url}}{{site.baseurl}}/clients/java/) instead.
+{: .warning}
 
 ## Legacy clients
 
 Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch, but the latest versions of those clients might include license or version checks that artificially break compatibility. This page includes recommendations around which versions of those clients to use for best compatibility with OpenSearch.
 
+OpenSearch 2.0.0 no longer supports compatibility with legacy clients. Due to breaking changes with REST APIs, some features are not supported when using OpenSearch 1.x clients to connect to OpenSearch 2.0.
```

Comment:
This is not true across the board. We'll need to double check this statement.

Reply:
I think its best to refer to the COMPATIBILITY.md in each client to get the latest compatibility matrix. Can we add something referring to that? Since it can be different for different clients?